client/src/routes/join/+page.svelte

```diff
@@ -54,18 +60,22 @@
       code = queryCode;
     }
   });
+
   async function joinLobby() {
     code = code.toLowerCase();
+    waiting = true;
     try {
       await saveOrCreate($user, userData, name.trim());
       // enter lobby with the user's info
       await findAndJoinLobby(code);
       // go to game page
       goto(`/game?code=${code}`);
     } catch (err) {
+      waiting = false;
       // if the lobby doesn't exist then error is thrown
       errorMessage = err instanceof Error ? err.message : String(err);
       if (errorMessage == "You are already in the lobby!") {
+        waiting = true;
```

Comment:
what does this do?

Reply:
If you are already in the lobby, the button should be disabled. Otherwise other errors being thrown should keep `waiting = false`